lib/matplotlib/ticker.py

```diff
@@ -1170,25 +1170,177 @@ class LogitFormatter(Formatter):
+
+        minor : bool (default: False)
+            indicate if the formatter is formatting minor ticks or not.
+            Basically minor ticks are not labelled, except when only few ticks
+            are provided, the most espaced ticks are labelled. See others
+            parameters to change the default behavior.
+
+        minor_threshold : int (default: 25)
+            maximum number of locs for labelling some minor ticks. This
+            parameter have no effect if minor is False.
+
+        minor_number : int (default: 6)
+            number of ticks which are labelled when the number of ticks is
+            below the threshold.
+        """
+        self._use_overline = use_overline
+        self._minor = minor
+        self._labelled = set()
+        self._minor_threshold = minor_threshold
+        self._minor_number = minor_number
+
+    def use_overline(self, use_overline):
+        r"""
+        Switch display mode with overline for labelling p>1/2.
+
+        Parameters
+        ----------
+        use_overline : bool (default: False)
+            if x > 1/2, with x = 1-v, indicate if x should be displayed as
+            $\overline{v}$. The default is to display $1-v$.
+        """
+        self._use_overline = use_overline
+
+    def set_minor_threshold(self, minor_threshold):
+        """
+        Set the threshold for labelling minors ticks
+
+        Parameters
+        ----------
+        minor_threshold : int
+            maximum number of locs for labelling some minor ticks. This
+            parameter have no effect if minor is False.
+        """
+        self._minor_threshold = minor_threshold
+
+    def set_minor_number(self, minor_number):
+        """
+        Set the number of minor ticks to label when some minor ticks are
+        labelled.
+
+        Parameters
+        ----------
+        minor_number : int
+            number of ticks which are labelled when the number of ticks is
+            below the threshold.
+        """
+        self._minor_number = minor_number
+
+    def set_locs(self, locs):
+        self.locs = locs
+        self._labelled.clear()
+
+        if not self._minor:
+            return None
+        if all(
+            is_decade(x, rtol=1e-7)
+            or is_decade(1 - x, rtol=1e-7)
+            or (is_close_to_int(2 * x) and int(np.round(2 * x)) == 1)
+            for x in locs
+        ):
+            # minor ticks are subsample from ideal, so no label
+            return None
+        if len(locs) < self._minor_threshold:
+            if len(locs) < self._minor_number:
+                self._labelled.update(locs)
             else:
-                s = '$1-{:.5f}$'.format(1-x)
+                # we do not have a lot of minor ticks, so only few decades are
+                # displayed, then we choose some (spaced) minor ticks to label.
+                alocs = np.array(locs)
+                diff = np.diff(-np.log(1 / alocs - 1))
```

Comment:
Not really familiar with logit formulae but what's this log(1/x-1) that appears again and again?

Reply:
In math, `logit(x) = log(x/(1-x))`, we can write in base `e` or in any base. For all this computation knowing the base is useless.

For computing, to avoid numerical error, some of us (as me) prefer use the equivalent formula: `logit(x) = -log(1/x-1)` we have smaller errors.

Reply:
For example:

```
In [1]: import numpy as np                                                                                                              

In [2]: logit1 = lambda x: -np.log(1/x -1)                                                                                              

In [3]: logit2 = lambda x: np.log(x/(1-x))                                                                                              

In [4]: expit = lambda x: 1/(1+np.exp(-x))                                                                                              

In [5]: logit1(expit(10)) - 10                                                                                                          
Out[5]: -1.2967404927621828e-13

In [6]: logit2(expit(10)) - 10                                                                                                          
Out[6]: 9.698908343125368e-13
```

Reply:
OK, now I think I got it: you want to put your minor ticks as far as possible from the major ticks (taking the spacing between the major ticks as secondary tiebreaker), is that correct?  If so please add a comment explaining that, I've been staring at these formulae for way too long...
Also does that even work once you start adding more than one minor tick per space? wouldn't you pick two "central" ticks in that case, rather than having them at ~1/3 each?
Finally (again assuming I understood what you're doing...) note that the rest of the code already assumes base 10, in which case... wouldn't the minor ticks always appear in the same order anyways?  Do you need to redo this ordering computation every time, or can you just hard-code the minor ticks order?  (Still not sure I fully understood what's happening here.)

Reply:
You have the idea, except the formatter do not know the space from the next major ticks but only the space from the next minor ticks, which is not a big issue, because the nearest minor ticks is the nearest for most the case (except for .9, which is anywar near .8).

Yes, precomputed ticks could do the job for most case, but in fact we have a lot of case to distinguish:
 - many decades < .1
 - some decades <.1 and other is .5
 - some decades <.1 and other is .9
 - .1 and .5
 - …

So I prefer use a automatic computation than precompute in many case. LogFormatter have precomputed values, but it doesn't have many case to distinguish.

Please note this computation is done only one time for all values, not for each.

Reply:
Thanks.
Given how much time it took me to understand the computation here, there should probably be some comments in the code explaining what's going on...

Reply:
I put more comments, let me know.

Reply:
thanks, just a couple of typos but looks much better :)

Reply:
Thanks, it was not a very easy point to explain.